packages/@aws-cdk/aws-s3-deployment/test/integ.bucket-deployment-signcontent.ts

```diff
@@ -0,0 +1,52 @@
+import * as path from 'path';
+import * as iam from '@aws-cdk/aws-iam';
+import * as s3 from '@aws-cdk/aws-s3';
+import * as cdk from '@aws-cdk/core';
+import * as integ from '@aws-cdk/integ-tests';
+import { Construct } from 'constructs';
+import * as s3deploy from '../lib';
+
+class TestBucketDeployment extends cdk.Stack {
+  constructor(scope: Construct, id: string, props?: cdk.StackProps) {
+    super(scope, id, props);
+
+    const bucket = new s3.Bucket(this, 'Destination', {
+      removalPolicy: cdk.RemovalPolicy.DESTROY,
+      autoDeleteObjects: true, // needed for integration test cleanup
+    });
+    const policyResult = bucket.addToResourcePolicy(
+      new iam.PolicyStatement({
+        effect: iam.Effect.DENY,
+        principals: [new iam.AnyPrincipal()],
+        actions: ['s3:PutObject'],
+        resources: [`${bucket.bucketArn}/*`],
+        conditions: {
+          StringNotLike: {
+            's3:x-amz-content-sha256': '????????????????????????????????????????????????????????????????',
+          },
+        },
+      }),
+    );
+    if (!policyResult.statementAdded || !policyResult.policyDependable) {
+      throw new Error('test error - unable to add policy statement');
+    }
+
+    const deployment = new s3deploy.BucketDeployment(this, 'Deployment', {
+      sources: [s3deploy.Source.asset(path.join(__dirname, 'my-website'))],
+      destinationBucket: bucket,
+      signContent: true,
+      retainOnDelete: false, // default is true, which will block the integration test cleanup
+    });
+    // the test is only valid if the resource policy is already in place
+    deployment.node.addDependency(policyResult.policyDependable);
```

Comment:
There’s no new policy requirement - all buckets support both signed and unsigned content uploads by default. Setting signContent to true does not necessarily imply that the bucket always requires signed content, just that this deployment construct should use a signed upload.

Reply:
Unfortunately with the flexibility of IAM policies, I think it is impossible to determine from the bucket policy whether signContent needs to be set automatically.

Reply:
Oh, I see. The comment is not saying that the test will fail without the bucket policy. It’s saying that the test can have a false positive deployment success if it doesn’t wait for the policy, because the policy is what restricts unsigned uploads :)